pyuvsim/simsetup.py

```diff
@@ -1324,6 +1388,58 @@ def uvdata_to_config_file(uvdata_in, param_filename=None, telescope_config_name=
         yaml.dump(param_dict, yfile, default_flow_style=False)
 
 
+def complete_uvdata(uv_in, inplace=False):
+    """Fill out all required parameters of a :class:~`pyuvdata.UVData` object such that
+    it passes the :func:~`pyuvdata.UVData.check()`.
+
+    Arguments
+    ---------
+    uv_in : :class:~`pyuvdata.UVData` instance
+        Usually an incomplete object, containing only metadata.
+    inplace : bool, optional
+        Whether to perform the filling on the passed object, or a copy.
+
+    Returns
+    -------
+    :class:~`pyuvdata.UVData` : filled/completed object (if `inplace` is `True`, it is
+        the modified input)
+    """
+    if not inplace:
+        uv_obj = copy.deepcopy(uv_in)
+    else:
+        uv_obj = uv_in
+
+    uv_obj.set_drift()
+    uv_obj.vis_units = 'Jy'
+
+    uv_obj.instrument = uv_obj.telescope_name
+    if uv_obj.lst_array is None:
+        uv_obj.set_lsts_from_time_array()
+    uv_obj.spw_array = np.array([0])
+    if uv_obj.Nfreqs == 1:
+        uv_obj.channel_width = 1.  # Hz
+    else:
+        uv_obj.channel_width = np.diff(uv_obj.freq_array[0])[0]
+    uv_obj.set_uvws_from_antenna_positions()
+    if uv_obj.Ntimes == 1:
+        uv_obj.integration_time = np.ones_like(uv_obj.time_array, dtype=np.float64)  # Second
+    else:
+        # Note: currently only support a constant spacing of times
+        uv_obj.integration_time = (np.ones_like(uv_obj.time_array, dtype=np.float64)
+                                   * np.diff(np.unique(uv_obj.time_array))[0] * (24. * 60**2))  # Seconds
+
+    # Clear existing data, if any
```

Comment:
These seems like an important feature to document: if you pass this function a UVData object with a data array, the data will be zeroed.

Reply:
I've added a comment to the docstring, but I also figured it's probably better to make this function private. It doesn't have any external uses. The only user access to it now is through `initialize_uvdata_from_keywords`, which doesn't start with any data in it anyway.

Maybe I should add a warning if it actually overwrites data? I can imagine a weird situation where that could happen...